docs/src/locators.md

```diff
@@ -746,6 +845,62 @@ You can locate in the same way as if the shadow root was not present at all.
   await Expect(page.Locator("x-details")).ToContainTextAsync("Details");
   ```
 
+## Locate within elements
+
+You can chain methods that create a locator, like [`method: Page.getByText`] or [`method: Locator.getByRole`], to narrow down the search to a particular part of the page.
+
+For example, consider the following DOM structure:
+
+```html
+<div data-testid='product-card'>
+  <span>Product 1</span>
+  <button>Buy</button>
+</div>
+
+<div data-testid='product-card'>
+  <span>Product 2</span>
+  <button>Buy</button>
+</div>
+```
+
+Locating by role `button` will return an error due to [strictness](#strictness) as there is more than one element with this role and name. 
+
+In this scenario we can locate an element within another element. First we create a locator called **product** and locate the product cards by testid. We then filter by text to find the product card with the text of "Product 2". We can then use this new locator called **product** and locate the button inside that product card and click it.
```

Comment:
Very nice explanation! Perhaps a numbered list would be easier to read? Up to you though.